elisp/haskell-ide-engine.el

```diff
@@ -184,21 +184,47 @@ association lists and count on HIE to use default values there."
 
 (defun haskell-ide-engine-handle-first-plugins-command (json)
   "Handle first plugins call."
-
   (let ((menu-items
          (mapcar
           (lambda (plugin)
             (cons (symbol-name (car plugin))
                   (mapcar
                    (lambda (command)
-                     (vector (cdr (assq 'ui_description command)) (list 'hie-run-command (symbol-name (car plugin)) (cdr (assq 'name command)))))
+                     (vector (cdr (assq 'ui_description command))
+                             (intern (concat "haskell-ide-engine-"
+                                             (symbol-name (car plugin))
+                                             (cdr (assq 'name command))))))
+                   (cdr plugin))))
+          (cdr (assq 'plugins json))))
+        (command-names
+         (mapcar
+          (lambda (plugin)
+            (cons (car plugin)
+                  (mapcar
+                   (lambda (command)
+                     (cdr (assq 'name command)))
                    (cdr plugin))))
           (cdr (assq 'plugins json)))))
     (setq haskell-ide-engine-plugins (cdr (assq 'plugins json)))
+    (haskell-ide-engine-create-all-commands command-names)
     (easy-menu-define hie-menu hie-mode-map
       "Menu for Haskell IDE Engine"
       (cons "HIE" menu-items))))
 
+(defun haskell-ide-engine-create-command (plugin command)
+  `(defun ,(intern (concat "haskell-ide-engine-" (symbol-name plugin) command)) ()
```

Comment:
Sounds like a good idea, I just went with what was already there. I would do this in a different pr, since it’s a large diff (we need to rename the file to hie.el to conform to the elisp naming conventions). which only clutters what is going on here.